### PR TITLE
Added support for user namespace.

### PR DESCRIPTION
**Pull request by srikalyan**
Please let me know if you need any changes.

### CODE REVIEW

**Comment by srikalyan** [comment]
@GordonTheTurtle Done.

File: docker/api/container.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def start(self, container, binds=None, port_bindings=None, lxc_conf=None,
               dns=None, dns_search=None, volumes_from=None, network_mode=None,
               restart_policy=None, cap_add=None, cap_drop=None, devices=None,
               extra_hosts=None, read_only=None, pid_mode=None, ipc_mode=None,
-              security_opt=None, ulimits=None):
+              security_opt=None, ulimits=None, user_namespace=None):
```
Comment (shin-) [inline docker/api/container.py]: We are not adding new parameters to `start()` - anything but the container ID is deprecated anyways.

File: docker/utils/utils.py [hunk 2]
```diff
@@ -620,7 +620,7 @@ def create_host_config(binds=None, port_bindings=None, lxc_conf=None,
                        device_write_bps=None, device_read_iops=None,
                        device_write_iops=None, oom_kill_disable=False,
                        shm_size=None, version=None, tmpfs=None,
-                       oom_score_adj=None):
+                       oom_score_adj=None, user_namespace=None):
```
Comment (shin-) [inline docker/utils/utils.py]: Let's stick close to the engine's names and use `userns_mode` instead.

**Comment by shin-** [comment]
Thank you for your contribution!

I have left a couple comments on the code, please take the time to address those when you can.

Also, @GordonTheTurtle is a bot, so he won't be able to reply to you :)

File: docker/utils/utils.py (hunk 2 above)
Comment (srikalyan) [inline docker/utils/utils.py]: Done.

**Comment by srikalyan** [comment]
@shin- ,Thank you for the quick review. I made the changes you have suggested. @GordonTheTurtle seems not be an intelligent bot to reply to my questions :p.

**Comment by shin-** [comment]
Great, thanks! One last thing before going forward is to add the parameter to the docs, specifically in `docs/hostconfig.md` - and also while you are at it, please squash your commits! :)

**Comment by srikalyan** [comment]
@shin-, added doc and squashed commit. Please let me know if you need anything else.

**Comment by shin-** [comment]
Thanks, LGTM!

**Pull request merged**